vm/src/dictdatatype.rs

```diff
@@ -644,14 +646,14 @@ impl<T: Clone> Dict<T> {
 
     pub fn pop_back(&self) -> Option<(PyObjectRef, T)> {
         let mut inner = &mut *self.write();
-        let (entry_idx, entry) = inner.entries[..inner.next_new_entry_idx]
-            .iter_mut()
-            .enumerate()
-            .rev()
-            .find_map(|(i, entry)| entry.take().map(|e| (i, e)))?;
+        let entry = loop {
+            let entry = inner.entries.pop()?;
+            if let Some(entry) = entry {
+                break entry;
+            }
+        };
```

Comment:
Haven't touched dict in a while so I don't remember exactly how it all works, but this is different behavior from before. I suppose there are probably thousands of invocations of this while running the test suite, so if it was an issue it would show up, but does this leave dangling indices in the indices vector or something? I'm not seeing obvious changes where that change would be handled or something so just wondering.

Reply:
`next_new_entry_idx` was a redundant value of `len()` before. this function was the only exception.
So using `next_new_entry_idx` outside of this function is same as using the `len()`.

I thought that  means I only need to concern about this function's behavior. That doesn't look have much chance to be changed due to the lock of the first line of the function.

Reply:
@eldpswp99 do you have any advice about this?

Reply:
I used `next_new_entry_idx` to maintain the dict deletion system - deleting makes index element to DUMMY on `pop_back`. 
And since `should_resize` function and `resize` function will handle about danling indices, I think this change would remain the same behaviour.